src/mine_loop.rs

```diff
@@ -427,23 +430,27 @@ pub(crate) async fn make_coinbase_transaction_stateless(
 /// coinbase amount, since the guesser fee fraction is guaranteed to be in the
 /// range \[0;1\].
 ///
-/// Returns an array of two TxOutputs: liquid first, timelocked second.
+/// Returns: Either the empty list, or two outputs, one immediately liquid, the
+/// other timelocked for three years, as required by the consensus rules.
+///
+/// # Panics
+///
+/// If the provided guesser fee fraction is not between 0 and 1 (inclusive).
 pub(crate) fn composer_outputs(
     coinbase_amount: NativeCurrencyAmount,
     composer_parameters: ComposerParameters,
     timestamp: Timestamp,
-) -> Result<[TxOutput; 2]> {
+) -> TxOutputList {
```

Comment:
why go from returning a Result and error(s) to not returning a Result and panicking?

Reply:
No behavior has changed. Only the function signature. The function could not fail, as the `lossy_f64_fraction_mul` function already asserts that `composer_parameters.guesser_fee_fraction()` is a fraction.